### ISSUE TITLE
error: the server doesn't have a resource type "machinedeployment"

### ISSUE THREAD

**Issue by Kube-ASY** [kubermatic/kubeone#447]
**What happened**:
I tried to delete a cluster with the following config:

```
apiVersion: kubeone.io/v1alpha1
kind: KubeOneCluster
name: demo
versions:
  kubernetes: "1.13.2"
cloudProvider:
  name: "none"
hosts:
- publicAddress: '1.2.3.4'
  privateAddress: '172.18.0.1'
  sshPort: 22 # can be left out if using the default (22)
  sshUsername: ubuntu
machineController:
  deploy: false
  provider: "none"
```

This failed with the following issue:
```
~/go/bin/kubeone reset config.yaml -v
INFO[08:22:07 CEST] Resetting kubeadm…
INFO[08:22:07 CEST] Destroying worker nodes…                      node=172.18.0.1
+ kubectl cluster-info
+ 1> /dev/null
+ kubectl annotate --all --overwrite node 'kubermatic.io/skip-eviction=true'
node/master0 annotated
+ kubectl delete machinedeployment -n kube-system --all
error: the server doesn't have a resource type "machinedeployment"
Error: failed to exec command: export "PATH=$PATH:/sbin:/usr/local/bin:/opt/bin"

set -xeu pipefail


if kubectl cluster-info > /dev/null; then
  kubectl annotate --all --overwrite node kubermatic.io/skip-eviction=true
  kubectl delete machinedeployment -n "kube-system" --all
  kubectl delete machineset -n "kube-system" --all
  kubectl delete machine -n "kube-system" --all

  for try in {1..30}; do
    if kubectl get machine -n "kube-system" 2>&1 | grep -q  'No resources found.'; then
      exit 0
    fi
    sleep 10s
  done

  echo "Error: Couldn't delete all machines!"
  exit 1
fi
: Process exited with status 1
```

**What is the expected behavior**:
Deletion proceeds.

**How to reproduce the issue**:
Try to delete a cluster when the following is set in config.yaml
```
machineController:
  deploy: false
  provider: "none"
```

**Anything else we need to know?**

**Information about the environment**:
KubeOne version (kubeone version): v0.6.0-6-g6407139
Operating system: CentOS Linux release 7.6.1810 (Core)
Provider you're deploying cluster on: Bare-metal
Operating system you're deploying on: CentOS Linux release 7.6.1810 (Core)

**Comment by kron4eg** [comment]
Hey, thanks for the report!

We already aware of this issue. 
Closing as duplicate of #398

**Comment by xmudrii** [comment]
@Kube-ASY Thanks for reporting the issue!

This is a known issue and we have a ticket open for that one - #398. Considering this issue has more details than #398, I'll leave it open and close another one. We'll get this fixed for v0.7.0 release, which will be released in the upcoming weeks. As a workaround, you can skip destroying workers with the `--destroy-workers=false` flag.

**Comment by kron4eg** [comment]
changing milestone to 0.8 as original issue was in it

**Comment by xmudrii** [comment]
Considering this issue should not take too long to fix, I think we should get it done for 0.7, so I'll move it back and assign it to myself.

/assign
